python_modules/libraries/dagster-dbt/dagster_dbt/cloud_v2/resources.py

```diff
@@ -255,16 +265,58 @@ def load_check_specs(
         ]
 
     def cli(
-        self, args: Sequence[str], dagster_dbt_translator: Optional[DagsterDbtTranslator] = None
+        self,
+        args: Sequence[str],
+        dagster_dbt_translator: Optional[DagsterDbtTranslator] = None,
+        context: Optional[AssetExecutionContext] = None,
     ) -> DbtCloudCliInvocation:
         """Creates a dbt cli invocation with the dbt Cloud client."""
+        dagster_dbt_translator = validate_opt_translator(dagster_dbt_translator)
         dagster_dbt_translator = dagster_dbt_translator or DagsterDbtTranslator()
+
+        client = self.get_client()
         workspace_data = self.fetch_workspace_data()
+        job_id = workspace_data.job_id
+        manifest = workspace_data.manifest
+
+        assets_def: Optional[AssetsDefinition] = None
+        with suppress(DagsterInvalidPropertyError):
+            assets_def = context.assets_def if context else assets_def
+
+        selection_args: list[str] = []
+        indirect_selection_args: list[str] = []
+        if context and assets_def is not None:
+            manifest, dagster_dbt_translator = get_manifest_and_translator_from_dbt_assets(
+                [assets_def]
+            )
+
+            indirect_selection = os.getenv(DBT_INDIRECT_SELECTION_ENV, None)
+
+            selection_args, indirect_selection_override = get_subset_selection_for_context(
+                context=context,
+                manifest=manifest,
+                select="fqn:*",
+                exclude=None,
+                dagster_dbt_translator=dagster_dbt_translator,
+                current_dbt_indirect_selection_env=indirect_selection,
+            )
+
+            # set dbt indirect selection if needed to execute specific dbt tests due to asset check
+            # selection
+            indirect_selection = (
+                indirect_selection_override if indirect_selection_override else indirect_selection
+            )
+            indirect_selection_args = (
+                [f"--indirect-selection {indirect_selection}"] if indirect_selection else []
+            )
+
+        full_dbt_args = [*args, *selection_args, *indirect_selection_args]
```

Comment:
In general don't love the copy pasta here. Can we consolidate the functions? Once you have the manifest and translator in scope it should all be the same as core right?

Reply:
That's fair - the logic is pretty similar, but the `full_dbt_args` are slightly different. That said, I've opened a ticket in Linear to make that change.